packages/hawtio/src/plugins/shared/tree/node.ts

```diff
@@ -73,6 +73,7 @@ export class MBeanNode implements TreeViewDataItem {
       this.icon = Icons.mbean
     }
 
+    if (this.parent) this.parent.id = this.parent.generateId(true)
```

Comment:
why do we need to update the parent id?  I think id should be updated in its own and children shouldn't do that.

I think there's more subtle issue in this bug. At some point in the past, we decided to allow two nodes (substantial mbean and mere folder) with the same path up to themselves (e.g. `org.xnio:type=Xnio,provider="nio",worker="XNIO-1"`). So we decided to add suffix `-folder` to id so that the mbean node and folder node are handled separately. But looking at `org.xnio:type=Xnio,provider="nio",worker="XNIO-2"`, it seems it's rather behaving as mbean and folder in a single node, which is not something we expect. We'd need to figure out why and fix it so that it can behave the same way as XNIO-1.
![image](https://github.com/hawtio/hawtio-next/assets/156692/3c3c247d-a2fd-41ff-8ae2-29dab173e855)

Reply:
This one: #313

Reply:
Ok. From the code I thought a node was a folder if it has children, as there are several lines that to know if something is a folder it just checks if it has children.
MMMMM, I think what it'd be best would be to do a refactor of the tree class. As I have seen also there are several ways to add nodes to a folder (camel tree uses adopt. Other places I think just create MBean then assign the children. From what I have seen, there is no way to create a folder mbean with (...children) as its parameter which would have the desired behaviour you want.